schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/spi/MessageEndpoint.java

```diff
@@ -0,0 +1,109 @@
+                JasKeyTransformer.class,
+                JasOrganizationTransformer.class,
+                JasPaymentTransformer.class,
+                JasRepeatDetailsTransformer.class,
+                JasSubscriptionTransformer.class,
+                JasTransactionTransformer.class,
+                JasUserLoginTransformer.class,
+                JasUserTransformer.class
+        },
+        auth = @ApiAuth(allowCookieAuth = AnnotationBoolean.TRUE /* todo: I don't know another way :-( */),
+        namespace = @ApiNamespace(ownerDomain = "jasify.com",
+                ownerName = "Jasify",
+                packagePath = ""))
+public class MessageEndpoint {
+    private final ContactMessageDao contactMessageDao = new ContactMessageDao();
+
+    @ApiMethod(name = "contactMessages.get", path = "contact-messages/{id}", httpMethod = ApiMethod.HttpMethod.GET)
+    public ContactMessage getContactMessage(User caller, @Named("id") Key id) throws NotFoundException, UnauthorizedException, ForbiddenException {
+        checkFound(id, "id == null");
+        mustBeAdmin(caller);
+        try {
+            return contactMessageDao.get(id);
+        } catch (EntityNotFoundException e) {
+            throw new NotFoundException(e.getMessage());
+        }
+    }
+
+    @ApiMethod(name = "contactMessages.query", path = "contact-messages", httpMethod = ApiMethod.HttpMethod.GET)
+    public List<ContactMessage> getContactMessages(User caller) throws UnauthorizedException, ForbiddenException {
+        mustBeAdmin(caller);
+        return contactMessageDao.getAll();
+    }
+
+    @ApiMethod(name = "contactMessages.remove", path = "contact-messages", httpMethod = ApiMethod.HttpMethod.DELETE)
+    public void removeContactMessage(User caller, @Named("id") final Key id) throws NotFoundException, UnauthorizedException, ForbiddenException, InternalServerErrorException, BadRequestException {
+        checkFound(id, "id == null");
+        mustBeAdmin(caller);
+
+        try {
+            TransactionOperator.execute(new ModelOperation<Void>() {
+                @Override
+                public Void execute(Transaction tx) throws EntityNotFoundException {
+                    contactMessageDao.get(id); //Throws not found if this ContactMessage doesn't exist
+                    contactMessageDao.delete(id);
+                    tx.commit();
+                    return null;
+                }
+            });
+        } catch (EntityNotFoundException e) {
+            throw new NotFoundException(e.getMessage());
+        } catch (ModelException me) {
+            throw new InternalServerErrorException(me.getMessage());
+        }
+    }
+
+    @ApiMethod(name = "contactMessages.send", path = "contact-messages", httpMethod = ApiMethod.HttpMethod.POST)
+    public void sendContactMessage(@SuppressWarnings("unused")User caller, ContactMessage message) throws UnauthorizedException, ForbiddenException, BadRequestException, NotFoundException {
```

Comment:
Would be nice to actually add the value of `caller` in case this is a logged in user.  This helps us know if this is an existing user, and you don't need the `@SuppressWarnings` anymore :-)